File: ts/example/sort/inheritance/Animal.ts
import { Node } from '../tree/Node';
import { String } from '@otterop/lang/String';

export class Animal {

    private _word : String;

    public constructor(word : String) {
        this._word = word;
    }

    public act() : String {
        return this._word;
    }

    public species() : String {
        return String.wrap("unknown");
    }
}

File: go/example/sort/inheritance/animal.go
package inheritance

import (
    lang "github.com/otterop/otterop/go/lang"
)


type Animal struct {
    word *lang.String
}




func AnimalNew(word *lang.String) *Animal {
    this := new(Animal)
    this.word = word
    return this
}

func (this *Animal) Act() *lang.String {
    return this.word
}

func (this *Animal) Species() *lang.String {
    return lang.StringWrap(lang.StringLiteral("unknown"))
}